aws-predictions-tensorflow/src/test/java/com/amplifyframework/predictions/tensorflow/asset/InputTokenizerTest.java

```diff
@@ -0,0 +1,114 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.tensorflow.asset;
+
+import android.content.Context;
+import android.content.res.AssetManager;
+
+import com.amplifyframework.predictions.PredictionsException;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test that input conversion to TensorFlow Lite tokens
+ * works as intended.
+ */
+public final class InputTokenizerTest {
+    private static final long LOAD_TIMEOUT_MS = 100;
+
+    private Context mockContext;
+    private AssetManager mockAssets;
+
+    /**
+     * Set up mock behavior before each test.
+     */
+    @Before
+    public void setUp() {
+        // Mock Android context and asset manager
+        mockContext = mock(Context.class);
+        mockAssets = mock(AssetManager.class);
+    }
+
+    /**
+     * Test that text tokenizer converts input text into a 2-D
+     * array of equivalent token values and pads the rest with 0's.
+     * @throws Exception if dictionary fails to load
+     */
+    @Test
+    @SuppressWarnings("MagicNumber") // word tokens
+    public void testInputTextTokenizer() throws Exception {
+        final CountDownLatch loaded = new CountDownLatch(1);
+        final AtomicReference<Map<String, Integer>> tokens = new AtomicReference<>();
+        final AtomicReference<PredictionsException> error = new AtomicReference<>();
+
+        final String inputText = "Where is the bathroom?";
+        final InputStream stream = new FileInputStream("src/test/resources/word-tokens.txt");
+
+        // Make mock context return mock asset manager
+        when(mockContext.getAssets()).thenReturn(mockAssets);
+
+        // Make mock asset manager return pre-determined input stream
+        when(mockAssets.open(anyString())).thenReturn(stream);
+
+        // Load!! (from mock assets)
+        TextClassificationDictionary dictionary = new TextClassificationDictionary(mockContext)
+                .onLoaded(
+                    onLoad -> {
+                        loaded.countDown();
+                        tokens.set(onLoad);
+                    },
+                    onError -> {
+                        loaded.countDown();
+                        error.set(onError);
+                    }
+                );
+        dictionary.load();
+        loaded.await(LOAD_TIMEOUT_MS, TimeUnit.MICROSECONDS);
```

Comment:
(This will go away with `Await`, too. But ___nit:___ `MILLISECONDS`! Not `MICROSECONDS`.)

Reply:
oh wow i guess I shouldn't have relied on autocomplete LOL

Reply:
This exact thing happens to me all the time. It's probably something IntelliSense could improve.